va/va.go

```diff
@@ -393,6 +393,11 @@ func (va *ValidationAuthorityImpl) fetchHTTP(ctx context.Context, identifier cor
 		// Intercept DialContext in order to connect to the IP address we
 		// select.
 		DialContext: dialer.DialContext,
+		// These match the values from DefaultRoundTripper in net/http.
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
```

Comment:
From the docs:
> // ExpectContinueTimeout, if non-zero, specifies the amount of
    // time to wait for a server's first response headers after fully
    // writing the request headers if the request has an
    // "Expect: 100-continue" header. Zero means no timeout and
    // causes the body to be sent immediately, without
    // waiting for the server to approve.
    // This time does not include the time to send the request header.

We don't send the `Expect: 100-continue` so it sounds like we have no reason to set this.

Reply:
Counter-point: setting it seems pretty harmless and if some confluence of stupid circumstances down the road results in this being relevant we wouldn't want no timeout.

Reply:
Counter-counter-point: this change is being made to fix a problem we are having now and have (presumably) pinpointed to a missing handshake timeout. Adding in a possible-preventative fix that we don't actually know what will do in the internals is just going to muddy the waters for the real fix that we know we need.

Reply:
@rolandshoemaker That's fair. I don't feel that strongly about it.

@jsha Do you have a preference?